extensions/openstack/openstack-impl/src/main/java/org/mqnaas/extensions/openstack/capabilities/impl/OpenstackHostAdministration.java

```diff
@@ -0,0 +1,182 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.io.IOException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jclouds.openstack.nova.v2_0.NovaApi;
+import org.jclouds.openstack.nova.v2_0.domain.Flavor;
+import org.jclouds.openstack.nova.v2_0.domain.Server;
+import org.jclouds.openstack.nova.v2_0.features.ServerApi;
+import org.mqnaas.clientprovider.api.client.IClientProviderFactory;
+import org.mqnaas.clientprovider.exceptions.EndpointNotFoundException;
+import org.mqnaas.clientprovider.exceptions.ProviderNotFoundException;
+import org.mqnaas.core.api.IAttributeStore;
+import org.mqnaas.core.api.IRootResource;
+import org.mqnaas.core.api.Specification;
+import org.mqnaas.core.api.Specification.Type;
+import org.mqnaas.core.api.annotations.DependingOn;
+import org.mqnaas.core.api.annotations.Resource;
+import org.mqnaas.core.api.exceptions.ApplicationActivationException;
+import org.mqnaas.extensions.openstack.capabilities.host.api.IHostAdministration;
+import org.mqnaas.extensions.openstack.jclouds.clientprovider.IJCloudsNovaClientProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Closeables;
+
+/**
+ * <p>
+ * Specific implementation of the {@link IHostAdministration} capability for Openstack virtual machines.
+ * </p>
+ * 
+ * <p>
+ * This capability does not store any information about the {@link Server}. Every getter method uses the Jclouds client in order to retrieve this
+ * information from OpenStack.
+ * </p>
+ * 
+ * @author Adrian Rosello Rey (i2CAT)
+ * 
+ */
+public class OpenstackHostAdministration implements IHostAdministration {
+
+	private static final Logger	log	= LoggerFactory.getLogger(OpenstackHostAdministration.class);
+
+	private NovaApi				novaClient;
+
+	@DependingOn
+	IAttributeStore				attributeStore;
+
+	@DependingOn
+	IClientProviderFactory		clientProviderFactory;
+
+	@Resource
+	IRootResource				resource;
+
+	public static boolean isSupporting(IRootResource resource) {
+		Specification resourceSpec = resource.getDescriptor().getSpecification();
+
+		return (resourceSpec.getType().equals(Type.HOST) && resourceSpec.getModel().equals("openstack"));
+	}
+
+	@Override
+	public void activate() throws ApplicationActivationException {
+		log.info("Initializing OpenstackHostAdministration capability for resource " + resource.getId());
+
+		try {
+			novaClient = clientProviderFactory.getClientProvider(IJCloudsNovaClientProvider.class).getClient(resource);
```

Comment:
Invoking getClient() and storing its result to be re-used in subsequent calls is generally discouraged, as getClient() may return a different client depending on the user making a request.
However, it may be the case a different behaviour is desired. Is this the case?